libpod/pod_api.go

```diff
@@ -96,36 +97,40 @@ func (p *Pod) Stop(cleanup bool) (map[string]error, error) {
 	// dependencies. Should we bother with this?
 
 	// Stop to all containers
-	for _, ctr := range allCtrs {
-		ctr.lock.Lock()
-
-		if err := ctr.syncContainer(); err != nil {
-			ctr.lock.Unlock()
-			ctrErrors[ctr.ID()] = err
-			continue
-		}
 
-		// Ignore containers that are not running
-		if ctr.state.State != ContainerStateRunning {
-			ctr.lock.Unlock()
-			continue
-		}
+	var wg sync.WaitGroup
+	for _, c := range allCtrs {
+		wg.Add(1)
```

Comment:
Maybe put this in the `go func()` right before `defer wg.Done()`? Purely stylistic, but I think it makes it a little more clear